src/helpers/template-helper.js

```diff
@@ -30,6 +43,32 @@ const rvaluesVarSubstition = async (subjects, templateContext) => {
   return subjects
 }
 
+const rvaluesVarSubstitionMiddleware = async (req, res, next) => {
+  if (['POST', 'PUT'].indexOf(req.method) > -1) {
+    const token = req.headers.authorization
+    let msvcEndpoint
+    let iofogListEndPoint
+    if (token) {
+      try {
+        msvcEndpoint = await MicroservicesController.getMicroservicesByApplicationEndPoint(req)
+        msvcEndpoint = msvcEndpoint.microservices
+        iofogListEndPoint = await FogController.getFogListEndPoint(req)
+        iofogListEndPoint = iofogListEndPoint.fogs
+      } catch (e) {
+        // Nothing to do, suppose the token has no permission to access. The is the case of agent

```

Comment:
This part is due to post/put from agent. Their token throw an access error for getMicroservicesByApplicationEndPoint and getFogListEndPoint

==> need to do this middleware only for application and microservices post.